src/signature.rs

```diff
@@ -41,3 +46,29 @@ impl SignatureUtil for GenericArray<u8, U64> {
         signature::verify(&signature::ED25519, peer_public_key, msg, sig).is_ok()
     }
 }
+
+pub struct GenKeys {
+    generator: RwLock<ChaChaRng>,
```

Comment:
Why wrap this with a RwLock?

Reply:
I believe this is necessary because when I am using `fill_bytes`, if two processes use the same RNG at the same time they'll grab random values and interrupt the sequence

Reply:
it works when I have a RwLock, and it doesn't work when I don't so I will caveat this by admitting it is an assumption

Reply:
Hmm, that lock is a big red flag and suggests the key generation may only appear deterministic because you're coincidently queuing up the write lock in the order that rayon is spinning up threads. I wonder if you could remove the `RwLock` if you changed that `into_par_iter()` in `gen_n_keys` to `into_iter()`